app/controllers/charges_controller.rb

```diff
@@ -109,23 +108,44 @@ def validate(seller_id:, line_items:, is_distribution:)
         raise InvalidLineItem, 'Amount must be at least $0.50 usd'
       end
 
-      unless is_distribution && seller.cost_per_meal.present? && amount % seller.cost_per_meal != 0
+      @campaign = if is_distribution.present?
+                    # TODO(justintmckibben): Delete this case when we start using campaign_id
+                    #                        in the frontend
+                    campaign = Campaign.find_by(
+                      seller_id: @seller.id,
+                      active: true,
+                      valid: true
+                    )
+                    unless campaign
+                      raise ActiveRecord::RecordNotFound, "Passed in is_distribution with no active campaign running for seller_id=#{seller_id}"
+                    end
+
+                    campaign
+                  elsif charge_params[:campaign_id].present?
```

Comment:
FYI this is an untested codepath. I'll follow up with a test for this + will add a seed file